cmake/kokkos_arch.cmake

```diff
@@ -571,7 +576,7 @@ IF (KOKKOS_ENABLE_HIP)
     COMPILER_SPECIFIC_FLAGS(
       DEFAULT -fgpu-rdc
     )
-    IF (NOT KOKKOS_CXX_COMPILER_ID STREQUAL HIPCC)
+    IF (NOT KOKKOS_CXX_COMPILER_ID STREQUAL HIPCC AND NOT KOKKOS_IMPL_AMDGPU_FLAGS)
```

Comment:
I thought we had agreed not to guard anything in that block?

Reply:
That's not how I understood it

Reply:
I would prefer if you dropped that guard.  Looks good other than that.